File: Personal/ComicAndPicture/ComicAndPicture/Views/ComicItemView.m
//
//  ComicItemView.m
//  ComicAndPicture
//
//  Created by MS on 16/3/11.
//  Copyright © 2016年 JFY. All rights reserved.
//

#import "ComicItemView.h"
#import "RecordManager.h"

@implementation ComicItemView

- (instancetype)initWithFrame:(CGRect)frame AndWithItems:(NSInteger)items {
    
    if ([super initWithFrame:frame]) {
        
        [self createItemsWithItems:items];
        
    }
    
    return self;
    
}

- (void)createItemsWithItems:(NSInteger)items {
    
    CGFloat xSpace = (WIDTH - 70 * 4) / 5;
    
    for (int i = 0; i < items; i++) {
        
        UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
        
        [btn setTitle:[NSString stringWithFormat:@"%d话", i + 1] forState:UIControlStateNormal];
        
        [btn setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
        
        [btn setImage:[UIImage imageNamed:@"yuanlaibiaoqian"] forState:UIControlStateSelected];
        
        btn.backgroundColor = [UIColor colorWithWhite:0.2 alpha:0.2];
        
        btn.frame = CGRectMake(xSpace + (70 + xSpace) * (i % 4) , (5 + 40) * (i / 4), 70, 40);
        
        btn.layer.cornerRadius = 5;
        
        btn.tag = 1000 + i;
        
        [btn addTarget:self action:@selector(click:) forControlEvents:UIControlEventTouchUpInside];
        
        [self addSubview:btn];
        
    }
    
    
}


- (void)click:(UIButton *)btn {
    
    [self.myDelegate clickComicItemWithTag:btn.tag - 1000];
    
}


@end

File: Personal/小项目/ComicAndPicture/ComicAndPicture/View/ComicItemView.swift
//
//  ComicItemView.swift
//  ComicAndPicture
//
//  Created by vcyber on 16/11/9.
//  Copyright © 2016年 vcyber. All rights reserved.
//

import UIKit

@objc public protocol ComicItemViewDelegate: NSObjectProtocol{
    @objc optional func clickItem(tag:NSInteger) -> Void
}

class ComicItemView: UIScrollView {
    
    weak open var itemViewDelegate: ComicItemViewDelegate?
    

    init(frame: CGRect, items: NSInteger) {
        super.init(frame: frame)
        self.showsHorizontalScrollIndicator = false
        self.contentSize = CGSize(width: SCREEN_WIDTH, height: CGFloat(items / 4 * 45))
        let xSpace: CGFloat = (SCREEN_WIDTH - 70 * 4) / 5
        for i in 0..<items {
            let btn: UIButton = UIButton(frame: CGRect(x: xSpace + (70 + xSpace) * CGFloat(i % 4), y: (5 + 40.0) * CGFloat(i / 4), width: 70.0, height: 40.0))
            btn.layer.cornerRadius = 5
            btn.layer.masksToBounds = true
            btn.tag = 1000 + i
            btn.setTitle(String(format: "第%d话", i + 1), for: .normal)
            btn.backgroundColor = UIColor(white: 0.2, alpha: 0.2)
            btn.addTarget(self, action: #selector(click(btn:)), for: .touchUpInside)
            self.addSubview(btn)
        }
    }
    
    func click(btn: UIButton) -> Void {
        itemViewDelegate?.clickItem?(tag: btn.tag - 1000)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    /*
    // Only override draw() if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func draw(_ rect: CGRect) {
        // Drawing code
    }
    */

}
